rotkehlchen/chain/arbitrum_one/modules/hyperliquid/balances.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.chain.arbitrum_one.modules.hyperliquid.constants import CPT_HYPER
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.externalapis.hyperliquid import Hyperliquid
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.arbitrum_one.decoding.decoder import ArbitrumOneTransactionDecoder
+    from rotkehlchen.chain.arbitrum_one.node_inquirer import ArbitrumOneInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HyperliquidBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            evm_inquirer: 'ArbitrumOneInquirer',
+            tx_decoder: 'ArbitrumOneTransactionDecoder',
+    ):
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            tx_decoder=tx_decoder,
+            counterparty=CPT_HYPER,
+            deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+        if len(addresses_with_deposits := list(self.addresses_with_deposits(products=None))) == 0:
+            return balances
+
+        hyperliquid = Hyperliquid()
```

Comment:
this does not make sense. if this is the only place you use it then just make it part of this class. Or initialize it at class constructor. Or better yet initialize it first time you need it.
```suggestion
        if self.hyperliquid_api is None:
            self.hyperliquid_api = HyperliquidAPI()
```

In that class you create a connection session, initialize a token and set a string. It makes no sense to redo that every time you query balances.